### PR TITLE
Add celebrate_winner method and fix game flow bug

### PR DESCRIPTION
**Pull request by mkrumholz**
## What does this change?

Game class, Turn class, some small syntax cleanup

## What does this fix?

Adds celebrate_winner method and fixes issues with the workflow caused by Turn class method structure.

## Is this a bug or a feature? 

feature and bug fixes

## How was this tested?

added test for method in Turn class, used user_input to tests other functions

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -28,63 +32,63 @@ def main_menu
...
   def take_turns(user, computer)
     display_boards(user, computer)
     while winner?(user, computer) == false
-      computer_turn = Turn.new(computer, user, :computer)
-      computer_turn.take_turn(computer, user, :computer)
+      computer_turn = Turn.new(user, computer, :computer)
+      computer_turn.take_turn(user, computer, :computer)
       display_boards(user, computer)
       break if winner?(user, computer) == true
-      user_turn = Turn.new(computer, user, :human)
```
Comment (suzkiee) [inline lib/game.rb:53]: Can we add spaces between the display boards?

File: lib/game.rb [hunk 2]
```diff
@@ -28,63 +32,63 @@ def main_menu
...
-    winner 
+    winner
+  end
+
+  def winner?(user, computer)
+    check_for_winner(user, computer) != nil
+  end
+
+  def display_boards(user, computer)
+    puts "=============COMPUTER BOARD============="
```
Comment (suzkiee) [inline lib/game.rb:88]: spaces here I guess

File: lib/game.rb [hunk 3]
```diff
@@ -28,63 +32,63 @@ def main_menu
...
+    puts computer.render(true)
```
Comment (suzkiee) [inline lib/game.rb:89]: take out true after we test the celebrate_winner methods

File: lib/turn.rb [hunk 4]
```diff
@@ -28,58 +28,36 @@ def user_shoots(computer)
   end
 
   def validate_user_shot(computer, coordinate)
-    while computer.valid_coordinate?(coordinate) == false
-      puts "Please enter a valid coordinate: "
+    while invalid?(computer, coordinate)  || already_fired_on?(computer, coordinate)
+      if invalid?(computer, coordinate)
```
Comment (suzkiee) [inline lib/turn.rb:32]: This was so awesome!

File: lib/game.rb (hunk 1 above)
Comment (mkrumholz) [reply lib/game.rb:53]: Added!

File: lib/game.rb (hunk 2 above)
Comment (mkrumholz) [reply lib/game.rb:88]: Done!

File: lib/turn.rb (hunk 4 above)
Comment (mkrumholz) [reply lib/turn.rb:32]: Yay!

**Pull request merged**